pages/proposals/proposals.jsx

```diff
@@ -0,0 +1,187 @@
+              this.setState({
+                submitting: true
+              }, () => {
+                this.submitProposal(this.state.formValues);
+              });
+            }
+          });
+        });
+      });      
+    });
+
+  },
+  submitProposal(proposal) {  
+    if (proposal.bilingual && proposal.bilingual == `Other` && proposal.bilingualother) {
+      proposal.bilingual = `Other: ${proposal.bilingualother}`;
+      delete proposal.bilingualother;
+    }
+
+    let request = new XMLHttpRequest();
+    request.open(`POST`, `/add-proposal`, true);
+    request.setRequestHeader("Content-type", "application/json");
+
+    request.onload = (event) => {
+      let resStatus = event.currentTarget.status;
+
+      this.setState({ 
+        submitting: false, 
+        submissionStatus: (resStatus >= 200 && resStatus < 400) ? SUBMISSION_STATUS_SUCCESS : SUBMISSION_STATUS_FAIL
+      });
+    };
+
+    request.onerror = (err) => {
+      console.log(err);
+    };
+
+    request.send(JSON.stringify(proposal));
+  },
+  renderForm() {
+    return (
+      <div className="content wide">
+        {this.renderIntro()}
+        <div className="form-section">
+          <Form ref="formPartOne" 
+            fields={fields.partOne}
+            inlineErrors={true}
+            onUpdate={this.handleFormUpdate} />
+        </div>
+        <div className="form-section">
+          <h1>Festival Spaces</h1>
+          <p>At MozFest, sessions are thematically organized into spaces. Please read the brief descriptions of each space <Link to="/spaces">here</Link> before answering the following questions.</p>
+          <Form ref="formPartTwo" 
+            fields={fields.partTwo}
+            inlineErrors={true}
+            onUpdate={this.handleFormUpdate} />
+        </div>
+        <div className="form-section">
+          <h1>Describe your session</h1>
+          <Form ref="formPartThree" 
+            fields={fields.partThree}
+            inlineErrors={true}
+            onUpdate={this.handleFormUpdate} />
+        </div>
+        <div className="form-section">
+          <h1>Additional support</h1>
+          <p>At MozFest, sessions are thematically organized into spaces. Please read the brief descriptions of each space <Link to="/spaces">here</Link> before answering the following questions.</p>
+          <Form ref="formPartFour" 
+            fields={fields.partFour}
+            inlineErrors={true}
+            onUpdate={this.handleFormUpdate} />
+        </div>
+        <button
+          ref="submitBtn" 
+          className="btn btn-primary-outline mr-3 my-5"
+          type="submit"
+          onClick={this.handleFormSubmit}
+          disabled={this.state.submitting ? `disabled` : null}
+        >{ this.state.submitting ? SUBMITTING_LABEL : PRE_SUBMIT_LABEL }</button>
+
+        { !this.state.submitting && this.state.submissionStatus === SUBMISSION_STATUS_FAIL && this.renderSubmissionFail() }
```

Comment:
Once `SUBMISSION_STATUS_FAIL` becomes true, this error never goes away, even if I were to fix my form and I get wifi back.